hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/BlocklistParser.java

```diff
@@ -0,0 +1,145 @@
+
+    /**
+     * todo
+     * Makes sure that all blocked accounts contained in the blocklist resource are present in state, and creates their definitions (if necessary).
+     *
+     * <p><b>Note: this method assumes that blocklists are enabled</b> – it does not check that config property
+     */
+    public List<BlockedInfo> parse(@NonNull final String blocklistResourceName) {
+        final List<String> fileLines = readFileLines(blocklistResourceName);
+        if (fileLines.isEmpty()) return Collections.emptyList();
+
+        return parseBlockList(fileLines);
+    }
+
+    private List<String> readFileLines(@NonNull final String blocklistResourceName) {
+        final List<String> fileLines;
+        try {
+            fileLines = readPrivateKeyBlocklist(blocklistResourceName);
+        } catch (Exception e) {
+            log.error("Failed to read blocklist resource {}", blocklistResourceName, e);
+            return Collections.emptyList();
+        }
+        return fileLines;
+    }
+
+    private static List<BlockedInfo> parseBlockList(final List<String> fileLines) {
+        final List<BlockedInfo> blocklist;
+        try {
+            final var columnHeaderLine = fileLines.get(0); // Assume that the first line is the header
+            final var blocklistLines = fileLines.subList(1, fileLines.size());
+            final var columnCount = columnHeaderLine.split(",").length;
+            blocklist = blocklistLines.stream()
+                    .map(line -> parseCSVLine(line, columnCount))
+                    .toList();
+        } catch (IllegalArgumentException iae) {
+            log.error("Failed to parse blocklist", iae);
+            return Collections.emptyList();
+        }
+        return blocklist;
+    }
+
+    @NonNull
+    private static List<String> readPrivateKeyBlocklist(@NonNull final String fileName) {
+        final var inputStream = BlocklistParser.class.getClassLoader().getResourceAsStream(fileName);
+        final var reader = new BufferedReader(new InputStreamReader(inputStream));
+        return reader.lines().toList();
+    }
+
+    /**
+     * Parses a line from the blocklist resource and returns blocked account info record.
+     *
+     * The line should have the following format:
+     * <private key>,<memo>
+     *     where <private key> is a hex-encoded private key
+     *     and <memo> is a memo for the blocked account
+     *     and both values are comma-separated.
+     *
+     * The resulting blocked account info record contains the EVM address derived from the private key, and the memo.
+     *
+     * @param line line from the blocklist resource
+     * @param columnCount number of comma-separated values in a line
+     * @return blocked account info record
+     */
+    @NonNull
+    private static BlockedInfo parseCSVLine(final @NonNull String line, final int columnCount) {
+        final var parts = line.split(",", -1);
```

Comment:
Did you copy this code from somewhere? I am too tired to review it thoroughly right now. 🙂

Reply:
Yes, everything in `BlocklistParser` was copied. I can spend the time to improve it now if needed, but I thought the time would be better spent elsewhere.